packages/jellyfish-api-core/__tests__/category/masternode/createMasternode.test.ts

```diff
@@ -253,8 +251,126 @@ describe('Multinodes masternodes', () => {
   it('should be failed as address is not owned by the wallet', async () => {
     const collateral1 = await group.get(1).getNewAddress()
 
-    const promise = clients[0].masternode.createMasternode(collateral1)
+    const promise = clientA.masternode.createMasternode(collateral1)
     await expect(promise).rejects.toThrow(RpcApiError)
     await expect(promise).rejects.toThrow(`Address (${collateral1}) is not owned by the wallet`)
   })
+
+  // timelock 10 -> 4, 5 -> 3, 0 -> 2
+  it('should createMasternode targetMultiplier checker', async () => {
+    const addrA0 = await clientA.wallet.getNewAddress('mnA0', AddressType.LEGACY)
+    const mnIdA0 = await clientA.masternode.createMasternode(addrA0)
+    await group.waitForMempoolSync(mnIdA0)
+
+    const addrA5 = await clientA.wallet.getNewAddress('mnA5', AddressType.LEGACY)
+    const mnIdA5 = await clientA.masternode.createMasternode(addrA5, '', { utxos: [], timelock: MasternodeTimeLock.FIVE_YEAR })
+    await group.waitForMempoolSync(mnIdA5)
+
+    const addrA10 = await clientA.wallet.getNewAddress('mnA10', AddressType.LEGACY)
+    const mnIdA10 = await clientA.masternode.createMasternode(addrA10, '', { utxos: [], timelock: MasternodeTimeLock.TEN_YEAR })
+    await group.waitForMempoolSync(mnIdA10)
+
+    const addrB0 = await clientB.wallet.getNewAddress('mnB0', AddressType.LEGACY)
+    const mnIdB0 = await clientB.masternode.createMasternode(addrB0)
+    await group.waitForMempoolSync(mnIdB0)
+
+    {
+      await group.get(0).generate(1)
+      await group.get(1).generate(1)
+      await group.waitForSync()
+
+      const mnA0 = (await clientA.masternode.getMasternode(mnIdA0))[mnIdA0]
```

Comment:
Looks off, I opened an issue on this matter https://github.com/DeFiCh/ain/issues/695#issuecomment-904620472